src/components/EventDatesComponent/index.js

```diff
@@ -0,0 +1,215 @@
+
+    wrapper.appendChild(link);
+    link.appendChild(img);
+    return wrapper;
+  }
+
+  _createCalendarIcon() {
+    const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+    svg.classList.add("icon-1x1-small");
+    svg.setAttribute("width", "24");
+    svg.setAttribute("height", "24");
+    svg.setAttribute("viewBox", "0 0 24 24");
+    svg.setAttribute("fill", "none");
+    svg.innerHTML = `
+          <g clip-path="url(#clip0_1327_21766)">
+            <path d="M17 3H21C21.2652 3 21.5196 3.10536 21.7071 3.29289C21.8946 3.48043 22 3.73478 22 4V20C22 20.2652 21.8946 20.5196 21.7071 20.7071C21.5196 20.8946 21.2652 21 21 21H3C2.73478 21 2.48043 20.8946 2.29289 20.7071C2.10536 20.5196 2 20.2652 2 20V4C2 3.73478 2.10536 3.48043 2.29289 3.29289C2.48043 3.10536 2.73478 3 3 3H7V1H9V3H15V1H17V3ZM20 9V5H17V7H15V5H9V7H7V5H4V9H20ZM20 11H4V19H20V11ZM6 13H11V17H6V13Z" fill="#191919"/>
+          </g>
+          <defs>
+            <clipPath id="clip0_1327_21766">
+              <rect width="24" height="24" fill="white"/>
+            </clipPath>
+          </defs>
+          `;
+    return svg;
+  }
+
+  _createClockIcon() {
+    const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+    svg.classList.add("icon-1x1-small");
+    svg.setAttribute("aria-hidden", "true");
+    svg.setAttribute("role", "img");
+    svg.setAttribute("viewBox", "0 0 24 24");
+    svg.setAttribute("fill", "currentColor");
+    svg.innerHTML = `
+          <title>Time icon</title>
+          <path d="M12 22C6.477 22 2 17.523 2 12C2 6.477 6.477 2 12 2C17.523 2 22 6.477 22 12C22 17.523 17.523 22 12 22ZM12 20C14.1217 20 16.1566 19.1571 17.6569 17.6569C19.1571 16.1566 20 14.1217 20 12C20 9.87827 19.1571 7.84344 17.6569 6.34315C16.1566 4.84285 14.1217 4 12 4C9.87827 4 7.84344 4.84285 6.34315 6.34315C4.84285 7.84344 4 9.87827 4 12C4 14.1217 4.84285 16.1566 6.34315 17.6569C7.84344 19.1571 9.87827 20 12 20ZM13 12H17V14H11V7H13V12Z"/>
+          `;
+    return svg;
+  }
+
+  _createNextButton() {
+    const link = this._createElement("a", "event-date_next-button");
+    link.href = "#";
+    const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+    svg.classList.add("icon-1x1-small");
+    svg.setAttribute("width", "24");
+    svg.setAttribute("height", "24");
+    svg.setAttribute("viewBox", "0 0 24 24");
+    svg.setAttribute("fill", "none");
+    svg.innerHTML = `
+          <g clip-path="url(#clip0_396_1289)">
+            <path d="M13.1722 12L8.22217 7.04999L9.63617 5.63599L16.0002 12L9.63617 18.364L8.22217 16.95L13.1722 12Z" fill="currentColor"></path>
+          </g>
+          <defs>
+            <clipPath id="clip0_396_1289">
+              <rect width="24" height="24" fill="white"></rect>
+            </clipPath>
+          </defs>
+          `;
+    link.appendChild(svg);
+    return link;
+  }
+
+  _addSeeMoreAndNextButtonIfNeeded(component) {
```

Comment:
We can do the same approach as `elementRef` to assign it to the class property. And we need to rename it from `component` to a bit more specific.